homeassistant/components/recorder/__init__.py

```diff
@@ -687,6 +702,8 @@ def _run_event_loop(self):
     def _process_one_event_or_recover(self, event):
         """Process an event, reconnect, or recover a malformed database."""
         try:
+            if self._process_one_task(event):
+                return
             self._process_one_event(event)
```

Comment:
The typical case is to record events, not handle special tasks.
Should we try that first?

Reply:
No we can't as that code does not check the type. It accesses `event.event_type`, which is not a valid property on tasks.

Btw, the only reason I separated this in a function was because a linter complained about function lengths (the two lines seemed to pushed it over a limit).

Reply:
Implemented in https://github.com/home-assistant/core/pull/61161